src/sign1.rs

```diff
@@ -173,6 +176,13 @@ impl CoseSign1 {
     pub fn payload(&self) -> Option<&ByteBuf> {
         self.inner.2.as_ref()
     }
+
+    pub fn claims_set(&self) -> Option<Result<ClaimsSet>> {
```

Comment:
in general I think it's better to return `Result<Option<...>, ...>` over `Option<Result<..., ...>>`, as most of the time call sites will expect to have `Result` as the outermost layer

Reply:
Makes sense--will update those return types!